public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiInMemoryTable, EuiSpacer, EuiLink, EuiFlyout, EuiButton } from "@elastic/eui";
+import _ from "lodash";
+import React, { useEffect, useContext, useState } from "react";
+import { SnapshotManagementService } from "../../../../services";
+import { CoreServicesContext } from "../../../../components/core_services";
+import { getErrorMessage } from "../../../../utils/helpers";
+import { GetIndexRecoveryResponse, CatSnapshotIndex } from "../../../../../server/models/interfaces";
+import { BREADCRUMBS } from "../../../../utils/constants";
+import { ContentPanel } from "../../../../components/ContentPanel";
+
+interface RestoreActivitiesPanelProps {
+  snapshotManagementService: SnapshotManagementService;
+  snapshotId: string;
+  repository: string;
+}
+
+export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId }: RestoreActivitiesPanelProps) => {
+  const context = useContext(CoreServicesContext);
+  const [startTime, setStartTime] = useState("");
+  const [stopTime, setStopTime] = useState("");
+  const [stage, setStage] = useState("");
+  const [indices, setIndices] = useState([{}]);
+  const [flyout, setFlyout] = useState(false);
+
+  useEffect(() => {
+    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
+    getRestoreStatus();
+  }, []);
+
+  const getRestoreStatus = async () => {
+    if (stage.indexOf("DONE") >= 0) {
+      console.log("done");
+      return;
+    }
+    try {
+      const res = await snapshotManagementService.getIndexRecovery();
+
+      if (res.ok) {
+        const response: GetIndexRecoveryResponse = res.response;
+
+        setRestoreStatus(response);
+        console.log(response);
+      } else {
+        context?.notifications.toasts.addDanger(res.error);
+      }
+    } catch (err) {
+      context?.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the recovery."));
+    }
+  };
+
+  const onIndexesClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFlyout(true);
+    console.log("index clicked");
+  };
+
+  const onCloseFlyout = () => {
+    setFlyout(false);
+  };
+
+  const setRestoreStatus = (response: GetIndexRecoveryResponse) => {
+    let minStartTime: number = 0;
+    let maxStopTime: number = 0;
+    let stageIndex: number = Infinity;
+    let doneCount: number = 0;
+    const indexes: CatSnapshotIndex[] = [];
+    const stages: string[] = ["START", "INIT", "INDEX", "FINALIZE", "DONE"];
```

Comment:
should we move this to some constants file?

Reply:
Looking at the usage below, can we instead use enum to define the different stages and then use the numerical values for stage comparisons?
something like

```
enum Stage {
  START = 0,
  INIT = 1,
  ...
  DONE = 4,
  INFINITY = Number.Infinity;
}

const stageOrder = [Stage.START, ....., Stage.DONE];
const currentStage: Stage = Stage.INFINITY;

...

doneCount = stage === Stage.DONE ? doneCount + 1 : doneCount;
...
```

Reply:
I will definitely look into this.  I am still working on getting better with Typescript.

Reply:
I've referred to stages.length instead of '4' for now.  this panel is still in progress for this PR.  perhaps your solution would be better applied to a fully working version of this panel.